apps/hash-graph/bin/cli/src/main.rs

```diff
@@ -3,29 +3,48 @@
 
 mod args;
 mod error;
+mod parser;
 mod subcommand;
 
+use std::sync::Arc;
+
 use error_stack::Result;
 use graph::load_env;
 
-use self::{args::Args, error::GraphError, subcommand::Subcommand};
+use self::{args::Args, error::GraphError};
 
-#[tokio::main]
-async fn main() -> Result<(), GraphError> {
+fn main() -> Result<(), GraphError> {
     load_env(None);
 
-    let args = Args::parse_args();
-
-    match args.subcommand {
-        Subcommand::Server(args) => subcommand::server(args).await,
-        Subcommand::Migrate(args) => subcommand::migrate(args).await,
-        Subcommand::TypeFetcher(args) => subcommand::type_fetcher(args).await,
-        Subcommand::Completions(ref args) => {
-            subcommand::completions(args);
-            Ok(())
-        }
-        Subcommand::Snapshot(args) => subcommand::snapshot(args).await,
-        #[cfg(all(hash_graph_test_environment, feature = "test-server"))]
-        Subcommand::TestServer(args) => subcommand::test_server(args).await,
+    let Args {
+        sentry_dsn,
+        subcommand,
+    } = Args::parse_args();
+
+    if let Some(dsn) = &sentry_dsn {
+        let client = sentry::Client::from_config(dsn);
+
+        assert!(client.is_enabled(), "Invalid Sentry DSN has been provided");
```

Comment:
Do we want to return an error instead? I see why you use `assert!` as this is not recoverable at this point, however, it's directly next to `main()` and an error has the IMO better output.

Reply:
fixed in [6e591ab](https://github.com/hashintel/hash/pull/3091/commits/6e591ab5bb600a6b7881131b437fb0e044584f83)